### DISCUSSION TITLE
Is the WebIDL spec for `BroadPhaseLayerInterfaceTable.MapObjectToBroadPhaseLayer()` correct?

### DISCUSSION THREAD

**Discussion by aSemy** [jrouwe/JoltPhysics.js#136] [General]
The `BroadPhaseLayer inBroadPhaseLayer` function parameter of `BroadPhaseLayerInterfaceTable.MapObjectToBroadPhaseLayer` is marked as `[Const, Ref]` in JoltJS.idl:

https://github.com/jrouwe/JoltPhysics.js/blob/5bf3e4e8b218c586e7519eb0834003d8908fc4f7/JoltJS.idl#L3328-L3331

However, in C++, inBroadPhaseLayer is not marked as const or a reference.

https://github.com/jrouwe/JoltPhysics/blob/bc9f6665c9f1912cb8ed4eeac44d09c2b70be883/Jolt/Physics/Collision/BroadPhase/BroadPhaseLayerInterfaceTable.h#L28-L32

For some context, I'm writing a pure-C wrapper generator using the WebIDL. Currently the C wrapper for MapObjectToBroadPhaseLayer fails to compile, unless I ignore the `[Const, Ref]` attributes. However, I can't ignore those attributes globally, because then the generated wrappers fail to compile.

Thanks!

**Comment by jrouwe** [comment]
Unfortunately it doesn't work without this, if I remove `[Const, Ref]` then the WebIDL binder script assumes that the object is passed by pointer so it generates:

```
void EMSCRIPTEN_KEEPALIVE emscripten_bind_BroadPhaseLayerInterfaceTable_MapObjectToBroadPhaseLayer_2(BroadPhaseLayerInterfaceTable* self, unsigned int inObjectLayer, BroadPhaseLayer* inBroadPhaseLayer) {
  self->MapObjectToBroadPhaseLayer(inObjectLayer, inBroadPhaseLayer);
}
```

Which doesn't compile.

I'm not sure why this would be an issue on your side as a function:

```
void function(BroadPhaseLayerInterfaceTable* self, unsigned int inObjectLayer, const BroadPhaseLayer& inBroadPhaseLayer) {
  self->MapObjectToBroadPhaseLayer(inObjectLayer, inBroadPhaseLayer);
}
```

should just work right? (it makes a copy of the object while calling)

**Reply by aSemy** [reply]
Thank you for the confirmation @jrouwe! I found a bug in my generator, and now the code is compiling.

Next step, figure out how to make the generated code completely C compatible...
